VocaDb.Migrations/Migrations.cs

```diff
@@ -1,17 +1,233 @@
+				.WithColumn("Status").AsString(10).NotNullable().WithDefaultValue("Draft")
 				.WithColumn("Address").AsString().NotNullable().WithDefaultValue(string.Empty)
+				.WithColumn("DefaultNameLanguage").AsString(20).NotNullable().WithDefaultValue("English")
 				.WithColumn("JapaneseName").AsString(255).NotNullable().WithDefaultValue(string.Empty)
 				.WithColumn("RomajiName").AsString(255).NotNullable().WithDefaultValue(string.Empty)
-				.WithColumn("EnglishName").AsString(255).NotNullable().WithDefaultValue(string.Empty);
+				.WithColumn("EnglishName").AsString(255).NotNullable().WithDefaultValue(string.Empty)
+				.WithColumn("AdditionalNamesString").AsString(1024).NotNullable().WithDefaultValue(string.Empty)
+				.WithColumn("Latitude").AsDouble().Nullable()
+				.WithColumn("Longitude").AsDouble().Nullable();
+
+			Create.Table("ArchivedVenueVersions")
+				.WithColumn("Id").AsInt32().NotNullable().PrimaryKey().Identity()
+				.WithColumn("Author").AsInt32().NotNullable().ForeignKey(TableNames.Users, "Id")
+				.WithColumn("ChangedFields").AsAnsiString(100).NotNullable()
+				.WithColumn("CommonEditEvent").AsAnsiString(30).NotNullable()
+				.WithColumn("Created").AsDateTime().NotNullable()
+				.WithColumn("Data").AsXml().NotNullable()
+				.WithColumn("Venue").AsInt32().NotNullable().ForeignKey(TableNames.Venues, "Id").OnDelete(Rule.Cascade)
+				.WithColumn("IsSnapshot").AsBoolean().NotNullable().WithDefaultValue(true)
+				.WithColumn("Notes").AsString(200).NotNullable()
+				.WithColumn("Status").AsString(10).NotNullable().WithDefaultValue("Draft")
+				.WithColumn("Version").AsInt32().NotNullable();
+
+			Create.Table("VenueNames")
+				.WithColumn("Id").AsInt32().NotNullable().PrimaryKey().Identity()
+				.WithColumn("Venue").AsInt32().NotNullable().ForeignKey(TableNames.Venues, "Id").OnDelete(Rule.Cascade)
+				.WithColumn("Language").AsString(16).NotNullable()
+				.WithColumn("Value").AsString().NotNullable();
+
+			Create.Table("VenueWebLinks")
+				.WithColumn("Id").AsInt32().NotNullable().PrimaryKey().Identity()
+				.WithColumn("Category").AsAnsiString(20).NotNullable()
+				.WithColumn("Description").AsString(512).NotNullable()
+				.WithColumn("Venue").AsInt32().NotNullable().ForeignKey(TableNames.Venues, "Id").OnDelete(Rule.Cascade)
+				.WithColumn("Url").AsString(512).NotNullable();
+				
+			Alter.Table(TableNames.EntryReports).AddColumn("Venue").AsInt32().Nullable().ForeignKey(TableNames.Venues, "Id").OnDelete(Rule.Cascade);
+			
+			Create.Column("Venue").OnTable(TableNames.ActivityEntries).AsInt32().Nullable()
+				.ForeignKey(TableNames.Venues, "Id").OnDelete(Rule.Cascade);
+
+			Create.Column("ArchivedVenueVersion").OnTable(TableNames.ActivityEntries).AsInt32().Nullable()
+				.ForeignKey(TableNames.ArchivedVenueVersions, "Id").OnDelete(Rule.None);
+
+			Create.Column("VenueEntry").OnTable(TableNames.AlbumReleaseEvents).AsInt32().Nullable().ForeignKey(TableNames.Venues, "Id").OnDelete(Rule.SetNull);

```

Comment:
I wanted to use the name `Venue`, but the name is already in use.

Reply:
I think it would be better to rename the old field to VenueName.

Reply:
If you rename the old field to `VenueName`, then you also have to rename the serialized `Venue` field in archived versions, so that revisions can be displayed correctly.

Reply:
It turned out to be no problem to rename Venue to VenueName in the AlbumReleaseEvents table, so I renamed it, but still the `Venue` field in the `ReleaseEventEditableFields` enum cannot be renamed easily because of the reason I said.

Reply:
I don't understand why. Besides, the field in ArchivedEventContract is already VenueName.

Reply:
The ChangedFields columns in the ArchivedEventVersions table are saved as comma-separated string. So if we rename `Venue` in the `ReleaseEventEditableFields` enum to `VenueName`, then we also have to update ChangedFields columns. I think we'll need to be careful with changes to existing table columns.

Reply:
Oh you mean that. It should be possible to update that SQL UPDATE command in the migration. In SQL server, REPLACE can be used to do string replace. So something like 

```sql
UPDATE [ArchivedEventVersions] SET (ChangedFields = REPLACE(ChangedFields, 'Venue', 'VenueName')) WHERE ChangedFields LIKE '%Venue%'
```

(above is not tested)

Reply:
Sorry, I was talking about the serialized XML at first, but it turned out that I was wrong.

Reply:
Actually, there's probably a FluentMigrator command for doing that update more easily too.